File: Chapter11/C#/SystemBrowser/Forms/FormMain.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace SystemBrowser
{
  /// <summary>
  /// FormMain hosts all the UI elements in the application,
  /// but doesn't manage any of them. The Binder wires event
  /// sources to handlers, without getting FormMain involved.
  /// </summary>
  public class FormMain : System.Windows.Forms.Form
  {
    #region Controls
    private System.ComponentModel.Container components = null;
    private System.Windows.Forms.Panel panelToolBar;
    private System.Windows.Forms.Panel panel1;
    private System.Windows.Forms.Panel panelStatusBar;
    private System.Windows.Forms.Panel panelNavigator;
    private System.Windows.Forms.Panel panelContent;
    #endregion

    #region UI Children
    public Control Toolbar
    {
      set 
      {
        value.Dock = DockStyle.Fill;
        panelToolBar.Controls.Add(value);
      }
    }

    public Control Statusbar
    {
      set 
      {
        value.Dock = DockStyle.Fill;
        panelStatusBar.Controls.Add(value);
      }
    }

    private Control navigatorFolders;
    public Control NavigatorFolders
    {
      set {navigatorFolders = value;}
    }

    private Control navigatorSearch;
    public Control NavigatorSearch
    {
      set {navigatorSearch = value;}
    }

    private Control contentFolders;
    private System.Windows.Forms.Splitter splitter1;
  
    public Control ContentFolders
    {
      set {contentFolders = value;}
    }

    private Control contentSearch;
    public Control ContentSearch
    {
      set {contentSearch = value;}
    }
    #endregion

    public FormMain()
    {
      // Required for Windows Form Designer support
      InitializeComponent();
    }

    public void ShowFolders()
    {
      panelNavigator.Controls.Clear();
      panelNavigator.Controls.Add(navigatorFolders);
      navigatorFolders.Dock = DockStyle.Fill;

      panelContent.Controls.Clear();
      panelContent.Controls.Add(contentFolders);
      contentFolders.Dock = DockStyle.Fill;
    }

    public void ShowSearch()
    {
      panelNavigator.Controls.Clear();
      panelNavigator.Controls.Add(navigatorSearch);
      navigatorSearch.Dock = DockStyle.Fill;

      panelContent.Controls.Clear();
      panelContent.Controls.Add(contentSearch);
      contentSearch.Dock = DockStyle.Fill;
    }
    
    /// <summary>
    /// Clean up any resources being used.
    /// </summary>
    protected override void Dispose( bool disposing )
    {
      if( disposing )
      {
        if (components != null) 
        {
          components.Dispose();
        }
      }
      base.Dispose( disposing );
    }

    #region Windows Form Designer generated code
    /// <summary>
    /// Required method for Designer support - do not modify
    /// the contents of this method with the code editor.
    /// </summary>
    private void InitializeComponent()
    {
      System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(FormMain));
      this.panelToolBar = new System.Windows.Forms.Panel();
      this.panel1 = new System.Windows.Forms.Panel();
      this.panelStatusBar = new System.Windows.Forms.Panel();
      this.panelNavigator = new System.Windows.Forms.Panel();
      this.splitter1 = new System.Windows.Forms.Splitter();
      this.panelContent = new System.Windows.Forms.Panel();
      this.SuspendLayout();
      // 
      // panelToolBar
      // 
      this.panelToolBar.Dock = System.Windows.Forms.DockStyle.Top;
      this.panelToolBar.Location = new System.Drawing.Point(0, 0);
      this.panelToolBar.Name = "panelToolBar";
      this.panelToolBar.Size = new System.Drawing.Size(770, 60);
      this.panelToolBar.TabIndex = 0;
      // 
      // panel1
      // 
      this.panel1.Location = new System.Drawing.Point(166, 3);
      this.panel1.Name = "panel1";
      this.panel1.Size = new System.Drawing.Size(539, 263);
      this.panel1.TabIndex = 2;
      // 
      // panelStatusBar
      // 
      this.panelStatusBar.Dock = System.Windows.Forms.DockStyle.Bottom;
      this.panelStatusBar.Location = new System.Drawing.Point(0, 398);
      this.panelStatusBar.Name = "panelStatusBar";
      this.panelStatusBar.Size = new System.Drawing.Size(770, 24);
      this.panelStatusBar.TabIndex = 1;
      // 
      // panelNavigator
      // 
      this.panelNavigator.Dock = System.Windows.Forms.DockStyle.Left;
      this.panelNavigator.Location = new System.Drawing.Point(0, 60);
      this.panelNavigator.Name = "panelNavigator";
      this.panelNavigator.Size = new System.Drawing.Size(222, 338);
      this.panelNavigator.TabIndex = 2;
      // 
      // splitter1
      // 
      this.splitter1.Location = new System.Drawing.Point(222, 60);
      this.splitter1.MinSize = 100;
      this.splitter1.Name = "splitter1";
      this.splitter1.Size = new System.Drawing.Size(6, 338);
      this.splitter1.TabIndex = 3;
      this.splitter1.TabStop = false;
      // 
      // panelContent
      // 
      this.panelContent.Dock = System.Windows.Forms.DockStyle.Fill;
      this.panelContent.Location = new System.Drawing.Point(228, 60);
      this.panelContent.Name = "panelContent";
      this.panelContent.Size = new System.Drawing.Size(542, 338);
      this.panelContent.TabIndex = 4;
      // 
      // FormMain
      // 
      this.AutoScaleBaseSize = new System.Drawing.Size(6, 15);
      this.ClientSize = new System.Drawing.Size(770, 422);
      this.Controls.Add(this.panelContent);
      this.Controls.Add(this.splitter1);
      this.Controls.Add(this.panelNavigator);
      this.Controls.Add(this.panelStatusBar);
      this.Controls.Add(this.panelToolBar);
      this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
      this.Name = "FormMain";
      this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
      this.Text = "SystemBrowser";
      this.ResumeLayout(false);

    }
    #endregion
  }
}

File: Chapter11/VB/SystemBrowser/Forms/FormMain.vb
Public Class FormMain
  Inherits System.Windows.Forms.Form

#Region "UI Children"
  Public WriteOnly Property Toolbar() As Control
    Set(ByVal Value As Control)
      Value.Dock = DockStyle.Fill
      panelToolBar.Controls.Add(Value)
    End Set
  End Property

  Public WriteOnly Property Statusbar() As Control
    Set(ByVal Value As Control)
      Value.Dock = DockStyle.Fill
      panelStatusBar.Controls.Add(Value)
    End Set
  End Property

  Private _navigatorFolders As Control
  Public WriteOnly Property NavigatorFolders() As Control
    Set(ByVal Value As Control)
      _navigatorFolders = Value
    End Set
  End Property

  Private _navigatorSearch As Control
  Public WriteOnly Property NavigatorSearch() As Control
    Set(ByVal Value As Control)
      _navigatorSearch = Value
    End Set
  End Property

  Private _contentFolders As Control
  Public WriteOnly Property ContentFolders() As Control
    Set(ByVal Value As Control)
      _contentFolders = Value
    End Set
  End Property

  Private _contentSearch As Control
  Public WriteOnly Property ContentSearch() As Control
    Set(ByVal Value As Control)
      _contentSearch = Value
    End Set
  End Property

#End Region

  Public Sub ShowFolders()
    PanelNavigator.Controls.Clear()
    PanelNavigator.Controls.Add(_navigatorFolders)
    _navigatorFolders.Dock = DockStyle.Fill

    PanelContent.Controls.Clear()
    PanelContent.Controls.Add(_contentFolders)
    _contentFolders.Dock = DockStyle.Fill
  End Sub

  Public Sub ShowSearch()
    PanelNavigator.Controls.Clear()
    PanelNavigator.Controls.Add(_navigatorSearch)
    _navigatorSearch.Dock = DockStyle.Fill

    PanelContent.Controls.Clear()
    PanelContent.Controls.Add(_contentSearch)
    _contentSearch.Dock = DockStyle.Fill
  End Sub

#Region " Windows Form Designer generated code "

  Public Sub New()
    MyBase.New()

    'This call is required by the Windows Form Designer.
    InitializeComponent()

    'Add any initialization after the InitializeComponent() call

  End Sub

  'Form overrides dispose to clean up the component list.
  Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
    If disposing Then
      If Not (components Is Nothing) Then
        components.Dispose()
      End If
    End If
    MyBase.Dispose(disposing)
  End Sub

  'Required by the Windows Form Designer
  Private components As System.ComponentModel.IContainer

  'NOTE: The following procedure is required by the Windows Form Designer
  'It can be modified using the Windows Form Designer.  
  'Do not modify it using the code editor.
  Friend WithEvents PanelToolBar As System.Windows.Forms.Panel
  Friend WithEvents PanelStatusBar As System.Windows.Forms.Panel
  Friend WithEvents PanelNavigator As System.Windows.Forms.Panel
  Friend WithEvents Splitter1 As System.Windows.Forms.Splitter
  Friend WithEvents PanelContent As System.Windows.Forms.Panel
  <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
    Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(FormMain))
    Me.PanelToolBar = New System.Windows.Forms.Panel
    Me.PanelStatusBar = New System.Windows.Forms.Panel
    Me.PanelNavigator = New System.Windows.Forms.Panel
    Me.Splitter1 = New System.Windows.Forms.Splitter
    Me.PanelContent = New System.Windows.Forms.Panel
    Me.SuspendLayout()
    '
    'PanelToolBar
    '
    Me.PanelToolBar.Dock = System.Windows.Forms.DockStyle.Top
    Me.PanelToolBar.Location = New System.Drawing.Point(0, 0)
    Me.PanelToolBar.Name = "PanelToolBar"
    Me.PanelToolBar.Size = New System.Drawing.Size(770, 60)
    Me.PanelToolBar.TabIndex = 1
    '
    'PanelStatusBar
    '
    Me.PanelStatusBar.Dock = System.Windows.Forms.DockStyle.Bottom
    Me.PanelStatusBar.Location = New System.Drawing.Point(0, 398)
    Me.PanelStatusBar.Name = "PanelStatusBar"
    Me.PanelStatusBar.Size = New System.Drawing.Size(770, 24)
    Me.PanelStatusBar.TabIndex = 2
    '
    'PanelNavigator
    '
    Me.PanelNavigator.Dock = System.Windows.Forms.DockStyle.Left
    Me.PanelNavigator.Location = New System.Drawing.Point(0, 60)
    Me.PanelNavigator.Name = "PanelNavigator"
    Me.PanelNavigator.Size = New System.Drawing.Size(222, 338)
    Me.PanelNavigator.TabIndex = 3
    '
    'Splitter1
    '
    Me.Splitter1.Location = New System.Drawing.Point(222, 60)
    Me.Splitter1.MinSize = 100
    Me.Splitter1.Name = "Splitter1"
    Me.Splitter1.Size = New System.Drawing.Size(6, 338)
    Me.Splitter1.TabIndex = 4
    Me.Splitter1.TabStop = False
    '
    'PanelContent
    '
    Me.PanelContent.Dock = System.Windows.Forms.DockStyle.Fill
    Me.PanelContent.Location = New System.Drawing.Point(228, 60)
    Me.PanelContent.Name = "PanelContent"
    Me.PanelContent.Size = New System.Drawing.Size(542, 338)
    Me.PanelContent.TabIndex = 5
    '
    'FormMain
    '
    Me.AutoScaleBaseSize = New System.Drawing.Size(6, 15)
    Me.ClientSize = New System.Drawing.Size(770, 422)
    Me.Controls.Add(Me.PanelContent)
    Me.Controls.Add(Me.Splitter1)
    Me.Controls.Add(Me.PanelNavigator)
    Me.Controls.Add(Me.PanelStatusBar)
    Me.Controls.Add(Me.PanelToolBar)
    Me.Icon = CType(resources.GetObject("$this.Icon"), System.Drawing.Icon)
    Me.Name = "FormMain"
    Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
    Me.Text = "SystemBrowser"
    Me.ResumeLayout(False)

  End Sub

#End Region

End Class
